File: c++/Disk.cpp
#include <iostream>
#include <string>

using namespace std;

class Disk {

    // deklarasi dan inisialisasi atribut private
    private:
        string type = "-";
        int capacity = 0;
        int price = 0;

    public:

        // constructor tanpa parameter
        Disk(){
            
        }

        // constructor dengan parameter
        Disk(string type, int capacity,int price){
            this->type = type;
            this->capacity = capacity;
            this->price = price;
        }

        
        // deklarasi setter sebagai method (Write)
        void setType(string type){
            this->type = type;
        }

        void setCapacity(int capacity){
            this->capacity = capacity;
        }

        void setPrice(int price){
            this->price = price;
        }

        
        // deklarasi getter sebagai method (Read Only)
        string getType(){
            return this->type;
        }

        int getCapacity(){
            return this->capacity;
        }

        int getPriceD(){
            return this->price;
        }

        // deklarasi method Display untuk menampilkan data menggunakan method Getter
        void displayDisk(){

            cout << "Disk Type      = " << this->getType() << endl; 
            cout << "Disk Capacity  = " << this->getCapacity() <<  " GB" << endl; 

        }

        // destructor
        ~Disk(){

        }

};

File: php/Disk.php
<?php
class Disk{
    
    // deklarasi dan inisialisasi atribut private
    private $type = "";
    private $capacity = 0;
    private $price = 0;


    // deklarasi setter sebagai method (Write)
    public function __construct($type = "", $cap = 0, $price = 0){
        $this->type = $type;
        $this->capacity = $cap;
        $this->price = $price;
    }
    

    // deklarasi getter sebagai method (Read Only)
    public function setType($type){
        $this->type = $type;
    }
    public function setCap($cap){
        $this->capacity = $cap;
    }
    public function setPriceD($price){
        $this->price = $price;
    }
    

    // deklarasi getter sebagai method (Read Only)
    public function getType(){
        return $this->type;
    }
    public function getCap(){
        return $this->capacity;
    }
    public function getPriceD(){
        return $this->price;
    }


    // deklarasi method Display untuk menampilkan data menggunakan method Getter
    public function displayDisk(){
        echo "Disk Type      = ". $this->getType(). "<br>";
        echo "Disk Capacity  = ". $this->getCap(). " GB". "<br>";
    }


    // destructor
    public function __destruct(){
    }
}

?>